worker.go

```diff
@@ -34,51 +36,66 @@ func (w *Worker) Start() {
 	w.running = true
 	defer func() { w.running = false }()
 
+	ctx, cancel := context.WithCancel(context.Background())
+
 	log := w.logentry
 	log.Info("starting")
 	for {
 		select {
+		case <-w.quit:
```

Comment:
Nope. We're creating the context in this same function, it can't be cancelled from anywhere else. The reason the context gets cancelled is that `w.quit` got a message, and when it does the execution ends, so there's no need to check `ctx.Done()`.